substrate/machine.py

```
@@ -0,0 +1,100 @@
+            inst = object.__new__(cls)
+            inst.id = id
+            inst.mode = None
+            inst.pull = -1
+            inst.calls = []
+            cls._instances[id] = inst
+            cls._levels.setdefault(id, 0)
+        if args or kwargs:
+            inst._configure(args, kwargs)
+        return inst
+
+    def __init__(self, id, *args, **kwargs):
```

Ruff (0.15.18)

[error] Function argument `id` is shadowing a Python builtin

(A002)